### ISSUE TITLE
Deleting a customer from the backend is not possible

### ISSUE THREAD

**Issue by azngeek** [PAYONE-GmbH/shopware-6#147]
As a user, it is not possible for me to delete a customer. The current database structure will result in an internal server error. Is this the expected behavior or could the table schema be modified?

```
CREATE TABLE `payone_payment_card` (
  `id` binary(16) NOT NULL,
  `customer_id` binary(16) NOT NULL,
  `pseudo_card_pan` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL,
  `truncated_card_pan` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL,
  `expires_at` datetime NOT NULL,
  `created_at` datetime(3) NOT NULL,
  `updated_at` datetime(3) DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `idx.payone_payment_card.expires_at` (`expires_at`),
  KEY `fk.payone_payment_card.customer_id` (`customer_id`),
  CONSTRAINT `fk.payone_payment_card.customer_id` FOREIGN KEY (`customer_id`) REFERENCES `customer` (`id`) ON DELETE RESTRICT ON UPDATE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```

**Comment by janteuber** [comment]
Hello,
That an error occurs here is not intended.
Where exactly is it possible to delete a customer in the SW6 backend?

**Comment by azngeek** [comment]
You can delete a customer by doing the following steps:

- Customers -> Overview
- Select any customer
- Select the three dots "..." and choose "Delete".

Due to GDPR as a merchant i need to be able to delete a customer on request and this is currently not possible because of that constraint.

![image](https://user-images.githubusercontent.com/213429/142036102-b493b707-5a74-4bab-a015-0c03890fdab5.png)

**Comment by janteuber** [comment]
Thanks for the explanation.
I was able to reproduce the bug and have already created a ticket for our agency.
In the next release, the bug should be fixed.

**Comment by janteuber** [comment]
Hello,
we solved that issue and create PR: https://github.com/PAYONE-GmbH/shopware-6/pull/149

**Comment by janteuber** [comment]
Finally we released it in v3.0.0
